### PR TITLE
OCPBUGS-3114: HyperShift: Do not accept empty infrastructure name

### PR DESCRIPTION
**Pull request by kyrtapz**
During CNO startup there is a small chance that it will read the infrastructure object before its `Status.InfrastructureName` is updated by hosted cluster config operator. 

To address such scenario do not accept empty InfrastructureName. 
Additionally add a retry to avoid unnecessary restarts.

Signed-off-by: Patryk Diak <pdiak@redhat.com>

### CODE REVIEW

File: pkg/operator/operator.go [hunk 1]
```diff
@@ -4,17 +4,17 @@ import (
 	"context"
 	"fmt"
 	configv1 "github.com/openshift/api/config/v1"
```
Comment (jcaamano) [inline pkg/operator/operator.go:6]: could you fix this import as well?

**Review by jcaamano — CHANGES_REQUESTED** [review]

File: pkg/operator/operator.go [hunk 2]
```diff
@@ -60,11 +61,23 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
...
+			}
+			return nil
+		})
+		if err != nil {
+			return fmt.Errorf("failed to get infrastructure name: %v", err)
 		}
 		cluster = infraConfig.Status.InfrastructureName
 	}
+
+	klog.Info("Creating status manager for %s cluster", cluster)
```
[the remark this answers is not in the document]
Comment (kyrtapz) [reply pkg/operator/operator.go:80]: https://github.com/openshift/cluster-network-operator/pull/1650

**Pull request merged**